src/controllers/projectCtrl.js

```diff
@@ -37,66 +37,100 @@ exports.deleteProject = (req, res) => {
 };
 
 exports.updateProjectInfos = (req, res) => {
-  Project.findByIdAndUpdate(req.params.projectId, req.body, {
-    new: true,
-  })
+  //check if project exist
+  Project.findById(req.params.projectId)
     .then((project) => {
-      if (!project)
-        res
-          .status(400)
-          .json({ message: 'No Item updated: The project does not exist' });
-      else
-        res.status(200).json({
-          message: 'The project has been modified correctly!',
-          data: project,
+      if (project) {
+        project = {
+          ...project,
+          ...req.body,
+        };
+        project
+          .save()
+          .then((newProject) =>
+            res.status(200).json({
+              message: 'Project infos updated!',
+              newProject,
+            })
+          )
+          .catch((error) => res.status(500).json(error));
+      } else {
+        res.status(404).json({
+          message: 'Project not found!',
         });
+      }
     })
     .catch((error) =>
       res.status(401).json({ message: 'Invalid Request!', error })
     );
 };
 
 exports.addUserToProject = (req, res) => {
-  Project.findByIdAndUpdate(
-    req.params.projectId,
-    {
-      $push: {
-        users: {
-          id: req.body.userId,
-          role: req.body.role,
-        },
-      },
-    },
-    {
-      new: true,
-      upsert: true,
-    }
-  )
+  //check if project exist
+  Project.findById(req.params.projectId)
     .then((project) => {
-      res.status(200).json({
-        message: 'The user has been added to the project!',
-        data: project,
-      });
+      if (project) {
+        const user = project.users.find(
+          (user) => user.toString() === req.body.userId
+        );
+        if (user) {
+          return res.status(400).json({
+            message: 'User already in the project',
+          });
+        }
+        project.users.push(req.body.userId);
+        project
+          .save()
+          .then((newProject) =>
+            res.status(200).json({
+              message: 'User added to project!',
+              newProject,
+            })
+          )
+          .catch((error) => res.status(500).json(error));
+      } else {
+        res.status(404).json({
+          message: 'Project not found!',
+        });
+      }
     })
     .catch((error) =>
       res.status(401).json({ message: 'Invalid Request!', error })
     );
 };
 
 exports.removeUserFromProject = (req, res) => {
-  Project.findByIdAndUpdate(
-    req.params.projectId,
-    { $pull: { users: req.body.userId } },
-    {
-      new: true,
-      upsert: true,
-    }
-  )
+  //check if project exist
+  Project.findById(req.params.projectId)
```

Comment:
Same as before

Reply:
Same, to check if user was in the team to send a Good error message (maybe there is a better way, I did this rapidly)